tests/ptvsd/common/test_launcher.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root
+# for license information.
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import os.path
+import pytest
+import subprocess
+import sys
+import time
+import ptvsd.common.launcher
+from ptvsd.common.launcher import parse, WAIT_ON_NORMAL_SWITCH, WAIT_ON_ABNORMAL_SWITCH
```

Comment:
For the imports, in new code, I've been sticking to two patterns to keep it readable, and avoid some issues we had in the past.

1. For modules from `ptvsd` and `tests`, consistently use `from ... import` - so `from ptvsd.common import launcher` here. I also insert a blank line between the modules that come from stdlib or site-packages (including non-standard pip-installed ones like `pytest`), and modules from src/ or tests/ (including `pydevd`) - just so that it's clear which is our code, and which is external dependencies.

2. Always import modules, never specific things from them. The reason is mainly because globals can be reassigned, and then you're stuck with the old value for one in any code that did `from ... import`. Most globals aren't reassigned; but the problem is that if that ever changes, the old code will quietly do the wrong thing. In addition, this also seems to work better in many cases where circular dependencies occur.

There is one exception to the last rule - `ptvsd.common.compat`. For that one, it's okay to do `from ptvsd.common.compat import bytes, unicode, xrange` etc; but specifically only for things that correspond to `builtins`. This is so that the rest of the code looks as if they really were built-in on all Python versions - i.e. `unicode(s)` rather than `compat.unicode(s)`.